starknet_devnet/blueprints/rpc/transactions.py

```diff
@@ -106,10 +106,10 @@ async def pending_transactions() -> List[RpcTransaction]:
 
 async def add_invoke_transaction(
     function_invocation: FunctionCall,
+    signature: List[Felt],
```

Comment:
`Optional` means the value can be `None` and is used when the default value is `None`. Parameters with a default value are defined last. The specification states that the signature is not a required parameter:
https://github.com/starkware-libs/starknet-specs/blob/v0.1.0/api/starknet_write_api.json#L23

I think the problem is also that JSON-RPC API v0.1.0 is not compatible with the new transaction version (1) and that this could be addressed with adaptation to API v0.2.0.
@drknzz what do you say, would it make sense to accept the solution in this PR as a temporary fix?

Reply:
Thank you,

The [addInvokeTransaction spec for v0.1 in RPC](https://github.com/starkware-libs/starknet-specs/blob/30e5bafcda60c31b5fb4021b4f5ddcfc18d2ff7d/api/starknet_write_api.json#L9) says the signature should come 2nd. The fact that it comes 5 after the Nonce is an issue for the Go SDK for the following reasons:
- Go only supports positional parameters, not named parameters
- Pathfinder have the parameter in the 2nd place already so I would need to implement different version of the protocol depending on the API provider.

I sure understand that in order to have the parameter optional, you need to push it to the end. Minimum I would prefer you put it in front of the Nonce that is not even supposed to exist in v0.1. I understand you are trying to make things better for everyone but, to me, leaving the signature in 2nd and making is mandatory (and possibly empty) is a better choice than pushing it in 5th. What do you think?

Reply:
There's some point in what @gregoryguillou is saying, so I think it could be accepted rn.

Mark that in RPC v0.2.0 the interface [changes](https://github.com/starkware-libs/starknet-specs/compare/v0.1.0...v0.2.0?diff=unified#diff-b10ba31e7dae56dbcf6f0b7d1c9d9e708b212a037c02b72741f276a04d9c14f4R11) though:

```
"name": "starknet_addInvokeTransaction",
"summary": "Submit a new transaction to be added to the chain",
"params": [
    {
        "name": "invoke_transaction",
        "description": "The information needed to invoke the function (or account, for version 1 transactions)",
        "required": true,
        "schema": {
            "$ref": "#/components/schemas/BROADCASTED_INVOKE_TXN"
        }
    }
]
```